command/setup_vault.go

```diff
@@ -0,0 +1,675 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package command
+
+import (
+	_ "embed"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+// Ensure SetupVaultCommand satisfies the cli.Command interface.
+var _ cli.Command = &SetupVaultCommand{}
+
+//go:embed asset/vault-wi-default-auth-method-config.json
+var vaultAuthConfigBody []byte
+
+//go:embed asset/vault-wi-default-policy.hcl
+var vaultPolicyBody []byte
+
+//go:embed asset/vault-wi-default-role.json
+var vaultRoleBody []byte
+
+const (
+	vaultRoleTasks  = "role-nomad-tasks"
+	vaultPolicyName = "policy-nomad-tasks"
+	vaultNamespace  = "nomad-workloads"
+	vaultAud        = "vault.io"
+)
+
+type SetupVaultCommand struct {
+	Meta
+
+	vClient  *api.Client
+	vLogical *api.Logical
+	ns       string
+
+	jwksURL string
+
+	cleanup bool
+	autoYes bool
+}
+
+// Help satisfies the cli.Command Help function.
+func (s *SetupVaultCommand) Help() string {
+	helpText := `
+Usage: nomad setup vault [options]
+
+  This command sets up Vault for allowing Nomad workloads to authenticate
+  themselves using Workload Identity.
+
+  This command requires acl:write permissions for Vault and respects
+  VAULT_TOKEN, VAULT_ADDR, and other Consul-related environment variables
```

Comment:
One thing that I find really annoying is that the default value if `VAULT_ADDR` is not set uses `https`.  Since this command is probably going to be used in local test environments would it make sense to special override this default to use `http`?